// appollo/index.js
import axios from "axios";
import { Mixin } from 'ts-mixer';
import Billings from "./resources/billing/index.js";
import Collection from "./resources/collections/index.js";
import Customers from "./resources/customers/index.js";
import Orders from "./resources/orders/index.js";
import Product from "./resources/products/index.js";
import Variants from "./resources/variants/index.js";


class Appollo extends Mixin(Product, Variants, Orders, Customers, Collection, Billings) {
    constructor(clientKey, clientSecret) {
        super()

        this.client_key = clientKey
        this.client_secret = clientSecret
        this.token = {}

        this.getAccessToken()
    }

    async getAccessToken() {
        try {
            const response = await axios.post(`https://api.tryappollo.com/admin/tokens/access`,
                {
                    client_key: this.client_key,
                    client_secret: this.client_secret
                })

            this.token = { ...response.data, created_at: Date.now() }
            return response.data
        } catch (error) {
            console.log(error.response);
        }
    }

    checkTokenExpiry() {
        const now = Date.now()
        const expiry = this.token.create_at + (this.token.expires_in * 1000)

        return expiry > now
    }

}


export default Appollo
